cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/DeterministicSelfHealingTest.java

```diff
@@ -0,0 +1,129 @@
+   *
+   * @param balancingConstraint Balancing constraint.
+   * @param cluster             The state of the cluster.
+   * @param goalNameByPriority  Name of goals by the order of execution priority.
+   * @param verifications The verifications to make.
+   */
+  public DeterministicSelfHealingTest(BalancingConstraint balancingConstraint,
+                                      ClusterModel cluster,
+                                      List<String> goalNameByPriority,
+                                      List<OptimizationVerifier.Verification> verifications) {
+    _balancingConstraint = balancingConstraint;
+    _cluster = cluster;
+    _goalNameByPriority = goalNameByPriority;
+    _verifications = verifications;
+  }
+
+  /**
+   * Populate parameters for the {@link OptimizationVerifier}.
+   *
+   * @return Parameters for the {@link OptimizationVerifier}.
+   */
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    Collection<Object[]> p = new ArrayList<>();
+
+    // Sorted by priority.
+    List<String> goalNameByPriority = Arrays.asList(RackAwareGoal.class.getName(),
+                                                    ReplicaCapacityGoal.class.getName(),
+                                                    DiskCapacityGoal.class.getName(),
+                                                    NetworkInboundCapacityGoal.class.getName(),
+                                                    NetworkOutboundCapacityGoal.class.getName(),
+                                                    CpuCapacityGoal.class.getName(),
+                                                    ReplicaDistributionGoal.class.getName(),
+                                                    PotentialNwOutGoal.class.getName(),
+                                                    DiskUsageDistributionGoal.class.getName(),
+                                                    NetworkInboundUsageDistributionGoal.class.getName(),
+                                                    NetworkOutboundUsageDistributionGoal.class.getName(),
+                                                    CpuUsageDistributionGoal.class.getName(),
+                                                    LeaderReplicaDistributionGoal.class.getName(),
+                                                    TopicReplicaDistributionGoal.class.getName());
+
+    Properties props = KafkaCruiseControlUnitTestUtils.getKafkaCruiseControlProperties();
+    props.setProperty(KafkaCruiseControlConfig.MAX_REPLICAS_PER_BROKER_CONFIG, Long.toString(5L));
+    BalancingConstraint balancingConstraint = new BalancingConstraint(new KafkaCruiseControlConfig(props));
+    balancingConstraint.setCapacityThreshold(TestConstants.LOW_CAPACITY_THRESHOLD);
+    balancingConstraint.setResourceBalancePercentage(TestConstants.LOW_BALANCE_PERCENTAGE);
+    List<OptimizationVerifier.Verification> verifications = Arrays.asList(DEAD_BROKERS, REGRESSION);
+
+    // ----------##TEST 1: SINGLE GOAL.
```

Comment:
Don't we already have the same tests with a single goal in `RandomSelfHealingTest`? If they are the same, can we remove the test with single goal and keep only the `RACK AWARE GOAL + ONE OTHER GOAL` part?
Finally, can we rename this test class to a more descriptive name -- e.g. `DeterministicSelfHealingTest` -> `SelfHealingWithOptimizedGoal`?